lib/sycamore/sycamore/connectors/elasticsearch/elasticsearch_reader.py

```diff
@@ -0,0 +1,70 @@
+from sycamore.data import Document
+from sycamore.connectors.base_reader import BaseDBReader
+from dataclasses import dataclass, field
+from typing import Dict
+
+from elasticsearch import Elasticsearch
+
+
+@dataclass
+class ElasticsearchReaderClientParams(BaseDBReader.ClientParams):
+    url: str
+    es_client_args: dict = field(default_factory=lambda: {})
+
+
+@dataclass
+class ElasticsearchReaderQueryParams(BaseDBReader.QueryParams):
+    index_name: str
+    query: Dict = field(default_factory=lambda: {"match_all": {}})
+    kwargs: Dict = field(default_factory=lambda: {})
```

Comment:
what goes in here?

Reply:
https://elasticsearch-py.readthedocs.io/en/v8.14.0/api/elasticsearch.html#elasticsearch.Elasticsearch.search

Several additional parameters that users might want to specify but are not required. I could remove `query` as well (and include as part of `kwargs`) but since it is required felt better to separate it out